src/irmin-graphql/server.ml

```diff
@@ -397,7 +398,7 @@ struct
   let _ = Lazy.force contents_as_node
 
   let err_write e =
-    Lwt.return_error (Irmin.Type.to_string Store.write_error_t e)
+    Lwt.return (Error (Irmin.Type.to_string Store.write_error_t e))
```

Comment:
Why such update is needed?

Reply:
See discussion [here](https://github.com/mirage/irmin/pull/974#issue-392484868). `Lwt` is still using the `result` compatibility package for these functions, which is annoying since we don't otherwise require it in Irmin.

Reply:
Strange ... but I understand.

Reply:
Would you be able to create an issue on that for Lwt @CraigFe? We should be able to remove it from Lwt and make 4.02 the base supported version.

Reply:
Done in https://github.com/ocsigen/lwt/issues/794.